fastapi/dependencies/utils.py

```diff
@@ -266,6 +267,79 @@ def get_typed_annotation(param: inspect.Parameter, globalns: Dict[str, Any]) ->
     return annotation
 
 
+def get_typed_signature_with_extra_parameters(call: Callable[..., Any]) -> inspect.Signature:
+    initial_params = get_typed_signature(call).parameters
+    extra_params = get_signature_extra_parameters(call).copy()
+
+    merged_params: List[inspect.Parameter] = []
+    for name, initial_param in initial_params.items():
+        extra_param = extra_params.pop(name, None)
+        if extra_param:
+            merged_param = merge_parameters(initial_param, extra_param)
+            merged_params.append(merged_param)
+        else:
+            merged_params.append(initial_param)
+    for extra_param in extra_params.values():
+        merged_params.append(extra_param)
+    merged_params.sort(key=lambda p: p.kind)
+    merged_params = filter(
+        lambda p: p.kind != inspect.Parameter.VAR_KEYWORD, merged_params
+    )
+    return inspect.Signature(merged_params)
+
+
+def get_signature_extra_parameters(
+    call: Callable[..., Any]
+) -> Dict[str, inspect.Parameter]:
+    return getattr(call, "__endpoint_extra_parameters__", dict())
```

Comment:
Use empty dict literal instead of the built-in function `dict`. It is faster.

Reply:
Yeah, I thought they were the same, but apparently not. ``{}`` is more optimized. Thanks for the hint)